src/hoc/ClientSideStateUpdate.tsx

```diff
@@ -87,6 +90,10 @@ export const ClientSideStateUpdate = ({
       const updatedTasks = tasksInStore.map((t) => (t.id === task.id ? task : t))
       store.dispatch(setTasks(updatedTasks))
     } //for updating a task in store with respect to task response from db in task details page
+
+    if (activities) {
+      store.dispatch(setActivities(activities))
+    }
```

Comment:
Why are we storing activities in a redux global state if this is only needed in a particular section?